assets/js/laws-regs-parser.js

```diff
@@ -242,3 +235,157 @@ function shareLink(e, divType) {
     shareIcon.style.display = 'block';
   }, 2000);
 }
+
+function addCopyEventListeners(btn, divType) {
+  btn.addEventListener('click', (e) => {
+    copyText(e, divType);
+  });
+  btn.addEventListener('mouseover', (e) => {
+    highlightText(e, divType);
+  });
+  btn.addEventListener('mouseleave', (e) => {
+    unHighlightText(e, divType);
+  });
+}
+
+function addPrintEventListeners(btn, divType) {
+  btn.addEventListener('click', (e) => {
+    printText(e, divType);
+  });
+  btn.addEventListener('mouseover', (e) => {
+    highlightText(e, divType);
+  });
+  btn.addEventListener('mouseleave', (e) => {
+    unHighlightText(e, divType);
+  });
+}
+
+function navResults(e, dir, prevBtn, nextBtn, currentCount, totalCount) {
+  e.preventDefault();
+  const count = parseInt(currentCount.innerText);
+  if (dir == 'next') {
+    if (count >= totalCount) {
+      return;
+    } else {
+      nextBtn.classList.remove('disabled');
+    }
+    const newCount = count + 1;
+    location.hash = '#result' + newCount;
+    currentCount.innerText = newCount;
+    if (newCount >= totalCount) {
+      nextBtn.classList.add('disabled');
+    }
+    if (newCount > 0) {
+      prevBtn.classList.remove('disabled');
+    }
+  } else {
+    if (count <= 0) {
+      return;
+    } else {
+      prevBtn.classList.remove('disabled');
+    }
+    const newCount = count - 1;
+    location.hash = '#result' + newCount;
+    currentCount.innerText = newCount;
+    if (newCount <= 0) {
+      prevBtn.classList.add('disabled');
+    }
+    if (newCount < totalCount) {
+      nextBtn.classList.remove('disabled');
+    }
+  }
+}
+
+function search() {
```

Comment:
this function is pretty large

I feel like there are a couple of different things going on here (highlighting, dealing with buttons)

Splitting those off might make it a bit more readable